source/merge-deep.d.ts

```diff
@@ -0,0 +1,47 @@
+/**
+Merge two types deeply into a new type. Keys of the second type overrides keys of the first type.
+
```

Comment:
> Write about some real-world use-cases where it can be useful. (It can be hard sometimes for users to see where they would use something) - https://github.com/sindresorhus/type-fest/blob/master/.github/contributing.md

Reply:
> If there has been any discussion somewhere about this type, include a link to it. For example, a discussion on the TypeScript issue tracker. - https://github.com/sindresorhus/type-fest/blob/master/.github/contributing.md